src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -113,6 +117,7 @@ public KNNQueryBuilder(StreamInput in) throws IOException {
             vector = in.readFloatArray();
             k = in.readInt();
             filter = in.readOptionalNamedWriteable(QueryBuilder.class);
+            ignoreUnmapped = in.readBoolean();
```

Comment:
Will this break in a mixed cluster state? i.e. some nodes have this code some do not?

Reply:
I'm not sure, but I don't see any downside to making it `readOptionalBoolean` instead.  I'll try the change and make sure it still fixes the bug.

Reply:
Unfortunately, I dont think `readOptionalBoolean` will resolve the issue. See https://github.com/opensearch-project/OpenSearch/blob/d2ca9261892ddc80c32d24870ee89b90e8916e9f/libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java#L598. What this will do is read a byte to check if the value is null. For it to work, the old node needs to write the byte, which it doesnt do.

@martin-gaievski  actually had a similar issue awhile ago and fixed it with something like this: https://github.com/opensearch-project/k-NN/blob/2.10/src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java#L120-L122

Reply:
Similar approach should work here for this change as well. We may want to extend that supported version and make it more like a map with a field name as kay and min version as a value, then check for each field on serialization/deserialization. 
for reference this is a PR with that change - https://github.com/opensearch-project/k-NN/pull/588